internal/cmd/logs/util.go

```diff
@@ -135,8 +135,12 @@ func getLogs(stackName, resourceName string) (events, error) {
 	// Filter out uninteresting messages
 	newLogs := make([]types.StackEvent, 0)
 	for _, log := range logs {
+		isLatestUserInitiated := log.ResourceStatusReason != nil && *log.ResourceStatusReason == "User Initiated"
 		if allLogs || (log.ResourceStatusReason != nil && !uninterestingMessages[*log.ResourceStatusReason]) {
 			newLogs = append(newLogs, log)
+			if sinceUserInitiated && isLatestUserInitiated {
+				break
```

Comment:
Does this rely on an implicit sort order for the logs? We don't sort them until after this code block.

Reply:
I see that this code relies on implicit sorting.
The getLogs function does not seem to be suitable for filtering logs because of the recursive processing involved.
Would it be better to filter the logs retrieved by using getLogs in logs.go?

Reply:
I moved the filter below sort function and create new slice.
Could you check please?

Reply:
Unit test is failing. The test currently only checks the output of the help text so it should be a simple fix. Ideally we would have more complete tests for this feature but it's not completely necessary, since there's no chance it would break the default behavior.

Reply:
I fixed unit test that outputs help text. The test passes successfully in the environment at local.